test/stdlib/ExtractElements.swift

```diff
@@ -0,0 +1,127 @@
+//===--- ExtractElements.swift - tests for lazy filtering-------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+
+let ExtractElements = TestSuite("ExtractElements")
```

Comment:
This file mimics the tests in `RemoveElements.swift`. Should we combine the two?